cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginClass.java

```diff
@@ -29,18 +30,18 @@
  */
 @Beta
 public class PluginClass {
-
   private final String type;
   private final String name;
   private final String description;
   private final String className;
   private final String configFieldName;
   private final Map<String, PluginPropertyField> properties;
   private final Set<String> endpoints;
+  private final Set<String> requirements;
 
   public PluginClass(String type, String name, String description, String className,
                      @Nullable String configfieldName, Map<String, PluginPropertyField> properties,
-                     Set<String> endpoints) {
+                     Set<String> endpoints, @Nullable Set<String> requirements) {
```

Comment:
it's better to avoid nullable collections when possible. Doesn't seem like this needs to be nullable.